static/app/views/explore/charts/index.tsx

```diff
@@ -44,14 +52,36 @@ export function ExploreCharts({query}: ExploreChartsProps) {
   const [dataset] = useDataset();
   const [visualizes, setVisualizes] = useVisualizes();
   const [interval, setInterval, intervalOptions] = useChartInterval();
+  const [groupBys] = useGroupBys();
+  const [resultMode] = useResultMode();
+  const topEvents = useTopEvents();
+
+  const fields: string[] = useMemo(() => {
+    if (resultMode === 'samples') {
+      return [];
+    }
+
+    return [...groupBys, ...visualizes.flatMap(visualize => visualize.yAxes)].filter(
+      Boolean
+    );
+  }, [resultMode, groupBys, visualizes]);
+  const [sorts] = useSorts({fields});
+
+  const orderby: string | string[] | undefined = useMemo(() => {
+    if (!sorts.length) {
+      return undefined;
+    }
+
+    return sorts.map(formatSort);
+  }, [sorts]);
```

Comment:
should you include `resultMode` in the memo? Actually is this that expensive a function we need to memoize in the first place?

Reply:
Hey, `resultMode` doesn't have to be a dependency since  `orderBy` is just a parsed `sort`. Also, `sort` is a string[], hence the memoization.